src/backend/proxy/providerPreload.ts

```diff
@@ -103,35 +107,46 @@ contextBridge?.exposeInMainWorld('providerApi', providerApi)
 
 function initProvider() {
   async function exposeWindowEthereum() {
-    console.log('exposing window ethereum')
-    if (!Object.hasOwn(window, 'ethereum')) {
-      const windowAny = window as any
-      windowAny.ethereum = {
-        request: windowAny.providerApi.provider.request,
-        send: windowAny.providerApi.provider.send,
-        sendAsync: windowAny.providerApi.provider.sendAsync,
-        once: windowAny.providerApi.provider.once,
-        on: windowAny.providerApi.provider.on,
-        off: windowAny.providerApi.provider.off,
-        addListener: windowAny.providerApi.provider.addListener,
-        removeListener: windowAny.providerApi.provider.removeListener,
-        removeAllListeners: windowAny.providerApi.provider.removeAllListeners,
-        isMetaMask: true,
-        enable: windowAny.providerApi.provider.enable,
-        selectedAddress: undefined,
-        accounts: undefined
-      }
-
-      windowAny.ethereum.on('accountsChanged', (accounts: string[]) => {
-        console.log('accounts changed', accounts)
-        windowAny.ethereum.selectedAddress = accounts[0]
-        windowAny.ethereum.accounts = accounts
-      })
+    const windowAny = window as any
+    windowAny.ethereum = {
```

Comment:
Since we have this object here and know the keys and values, maybe we could generate a interface from it and remove the `any` type.

Reply:
Sounds good. `window.ethereum` can be kind of hard to type but I took some time to improve our approach here. We now type it in one spot across the app using `MetaMaskInpageProvider` from `@metamask/providers`

I didn't have time to fix every type mismatch it threw (e.g. saying `selectedAddress` is read only when it is not) but this is a step in the right direction